colander/__init__.py

```diff
@@ -1184,7 +1184,10 @@ def serialize(self, node, appstruct):
                 else:
                     result = text_type(appstruct)
             else:
-                result = text_type(appstruct)
+                if self.encoding:
+                    result = text_type(appstruct).encode(self.encoding)
+                else:
+                    result = text_type(appstruct)
```

Comment:
I prefer that the `result = text_type(appstruct)` to be outside the `if`, with no `else`.  E.g.:

```python
    result = text_type(appstruct)
    if self.encoding:
        result = result.encode(self.encoding)
    return result
```